x-pack/plugins/uptime/public/hooks/use_kibana.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CoreStart } from '../../../../../src/core/public';
+import { KibanaReactContextValue, useKibana } from '../../../../../src/plugins/kibana_react/public';
+import { ClientPluginsStart } from '../apps/plugin';
+
+export type PluginKibanaContextValue = CoreStart & ClientPluginsStart;
+
+export const useKibanaContextForPlugin =
+  useKibana as () => KibanaReactContextValue<PluginKibanaContextValue>;
```

Comment:
This hooks seems useful that it's typed safe according to the plugin service, though not sure, that's worth the effort?

we usually use `useKibana` directly and then provide type into it. 

I am neutral about this.  though within uptime plugin this creates inconsistency that in most places we are using `useKibana` and here `useKibanaContextForPlugin`. So kind of makes it hard to read for new people jumping into the code, without context.

Reply:
@shahzad31 I agree with your comment that it creates inconsistency in the uptime plugin. I tried to use `useKibana`, but when I tried to use `observability`, which is part of the start dependencies for the uptime plugin, I was getting undefined. I might have done something wrong though. I will try to use `useKibana` again and see if the undefined error still persists. If yes, I will use my current solution for now and will come back to it once this PR is merged. This PR needs to be merged asap, since it is a prerequisite for other PRs.

Reply:
@shahzad31 Ok I switched back to `useKibana` and now I recall that I was not getting an undefined error but a Typescript error

```
Property 'observability' does not exist on type 'Partial<CoreStart>'.
```

That was the reason I introduced this new hook. Do you have an example on how you provide type to it, so that I am consistent with how you do it?